nav2_mppi_controller/src/critics/path_follow_critic.cpp

```diff
@@ -34,15 +34,16 @@ void PathFollowCritic::initialize()
 
 void PathFollowCritic::score(CriticData & data)
 {
-  if (!enabled_ ||
+  const size_t path_size = data.path.x.shape(0) - 1;
+
+  if (!enabled_ || path_size == 0 ||
```

Comment:
```suggestion
  if (!enabled_ || data.path.x.shape(0) == 0 ||
```
This leaves me a little uneasy. Can we not ask the above directly? Just in case `data.path.x.shape(0)` is ever 0 and we risk still entering because of the above subtraction? (I do not know what's going on in `xt` though)


Also, I like how it is done [here](https://github.com/ros-planning/navigation2/blob/8d4f6f4e2dcd37afae94d74e7b5b806ea78e9813/nav2_mppi_controller/src/critics/goal_angle_critic.cpp#L38) now as it a little better readable for non-C/C++'er used to short circuit evaluation, but nothing to worry over now :+1:

Reply:
https://github.com/botsandus/navigation2/pull/16